### DISCUSSION TITLE
automatic curve change when starting a game

### DISCUSSION THREAD

**Discussion by ghost** [Rem0o/FanControl.Releases#1711] [Ideas]
Id love to see a feature where your curve changes to a different curve when starting up a specific game. For Example my fans go to 100%  when my cpu reaches 85 degrees, but when i play valorant i use my headset and i dont hear my pc so my fans could go to 100% earlier. Having an option where Fan Control automatically changes a fan curve when starting up this or that game would be nice.

**Comment by leftcoastfunk** [comment]
+1 ... but better yet have it change *profile* rather than a single curve since many people use multiple different curves with different profiles set up for different scenarios

**Comment by leftcoastfunk** [comment]
In the absence of automatic curve/profile switching based on application start, I'd be happy with being able to assign keyboard shortcut(s) to different profiles. I have done this as a temporary workaround by creating windows shortcuts with the command line modifiers, but it's not the most ideal way of implementing it. I think it should be native to the app

**Comment by UrmasK222** [comment]
**Bare with me here. I have an alternative solution propisal in the end.**

**Question/Discussion**
Can't you just set up a single profile so that it keeps your PC cool and quiet with less demanding applications and pushes the fans up with higher demanding games? One curve could easily do it. In the end you want to keep your hardware cooled regardless of what you are doing. I mean.. playing Sims with CPU at 90 degrees is not an option, is it?  Either is it logical to push fans to 100% because Sims probably only needs fanless AIO :D (unless you have uncapped FPS and your RTX3090 runs it at 1000FPS).

The profiles can be, for example - "Quiet gaming", "Cool gaming" etc. So if it's "Cool gaming" it always pushes fans to 100% but doesn't do that before, say, GPU reaches 65 degrees and CPU reaches 70 degrees etc. So browsing web, working is still quiet. 

And "Quiet gaming" never pushes over, say 60%, even while gaming but it still has to have a critical point where all fans go up to 100% pull down the temps from GPU 88 and CPU 95 or whatever the limits are there. You can also consider RAM, VMOS, SSD etc. for critical situations, a back-up so to say.

**And now let's combine these.**

**Personal example. Solution based on one profile.**

I have set it up so that my fans usually run at 60%, not more. In my case, if a GPU fan failure or some other external force majeure, if the GPU hot spot reaches 99 degrees*, it pushes all fans to 100%. This has never happened but I have tested it and it works. I run it quite quiet during gaming but I have the emergency profile set into my quiet profile. 

And also, fyi, I have limited my World of Tanks FPS to my monitor refresh rate, 165Hz, which keeps GPU a bit cooler because high FPS is more demanding (try it). Sims can be easly 100FPS/Hz etc.

*_As per AMD info I've found, the 6700XT maximum hot spot (junction temperature) is 110 degrees. That's my reasoning for 99 degrees_

**A picture tells more than all my words:**
![FanControl quiet profile](https://user-images.githubusercontent.com/45017739/234261653-0ad8e005-2fe1-41db-9c8d-1b1589143492.png)

**Final takes**
It works great for me and I can put all my needs into one profile, I believe. 

For Call of Duty games I still have to include a set up there because it's much more demanding overall. I just created this previously mentioned profile so It will take some time and testing to get it right ;) But I intend to put everything into one profile with the exemption of summer vs winter time profiles where summer time will have more fans running in idle mode at low speeds. 

The only thing that I can think of is the fact that an update (drivers, FanControl) might break this profile. So I am saving images of the setups I have, just in case ;).

Br

**Reply by UrmasK222** [reply]
Actually, this was easier than I thought. 

I checked my maximum GPU hot spot temp for today and it was 92 degrees. So I added COD edition to my profile and set it to 93 degrees (for testing). I can push it up a little if I want to.

I'm almost sure that COD MW2 or Warzone push the internals of my PC case so high that the GPU hot spot temp will rise also. So then fans start spinning at 80%. GPU fans will take care of themselves. ;) So it's 60% up to 92 degrees, 80% up to 98 degrees and 100% starting from 99 degrees. This is based on GPU hotspot only, for now.

If taken time, I could replace the "Trigger" fan curves with manual fan curves (just like I did for CPU) but I don't have any need for that right now.

![cod edition](https://user-images.githubusercontent.com/45017739/234266768-bba44fdf-e553-46b3-9cec-cf8c5576582d.png)
